### PR TITLE
fix(typo): unqualifed to unqualified

### PR DESCRIPTION
**Pull request by waynexia**
## Which issue does this PR close?

Closes #.

## Rationale for this change

I find (actually this is found by my typo ci) this typo "unqualifed" in our public API.

## What changes are included in this PR?

Correct the typo

## Are there any user-facing changes?

Yes, this is a breaking change.

I'm not sure how we should handle this. Maybe add a new correct spelling method that proxies to the current implementation, and deprecate the typo one?

Further more, is it necessary to setup a typo CI in this repository as well 🤔

### CODE REVIEW

**Comment by alamb** [comment]
Thanks @waynexia 

> I'm not sure how we should handle this. Maybe add a new correct spelling method that proxies to the current implementation, and deprecate the typo one?

I think that would be the most user friendly option if you can do so

**Comment by waynexia** [comment]
> Thanks @waynexia
> 
> > I'm not sure how we should handle this. Maybe add a new correct spelling method that proxies to the current implementation, and deprecate the typo one?
> 
> I think that would be the most user friendly option if you can do so

Great. I removed it in [c579528](https://github.com/apache/datafusion/pull/11159/commits/c579528d1b57628117c10fc47a8e62e1dc22fbb8)

File: datafusion/common/src/dfschema.rs [hunk 1]
```diff
@@ -159,6 +159,19 @@ impl DFSchema {
...
+    pub fn from_unqualified_fields(
+        fields: Fields,
+        metadata: HashMap<String, String>,
+    ) -> Result<Self> {
+        Self::from_unqualifed_fields(fields, metadata)
+    }
+
+    /// Create a new `DFSchema` from a list of Arrow [Field]s
+    #[deprecated(
+        since = "40.0.0",
```
Comment (comphead) [inline datafusion/common/src/dfschema.rs:172]: 👍 wow what a typo

File: datafusion/common/src/dfschema.rs [hunk 2]
```diff
@@ -159,6 +159,19 @@ impl DFSchema {
...
+        note = "Please use `from_unqualified_fields` instead (this one's name is a typo)"
```
Comment (comphead) [inline datafusion/common/src/dfschema.rs:173]:
```suggestion
        note = "Please use `from_unqualified_fields` instead (this one's name is a typo). This method is subject to be removed soon"
```

**Review by alamb — APPROVED** [review]
Thank you @waynexia and @comphead

**Pull request merged**